src/phase0/state_transition.rs

```diff
@@ -0,0 +1,48 @@
+use crate::phase0::beacon_block::SignedBeaconBlock;
+use crate::phase0::beacon_state::BeaconState;
+
+pub fn apply_block<
```

Comment:
this is probably the worst it will get, so if we can handle this, then i'd say full-speed ahead :)